### ISSUE TITLE
self._job_name is confusing

### ISSUE THREAD

**Issue by davidmarin** [Yelp/mrjob#982]
There's an instance variable self._job_name, which is meant to be unique for each job. We use it, for example, to pick the name of the job's temp directory.

This seems like it ought to be the same as the job name in Hadoop (`mapreduce.job.name`), but it's not. This led to a ticket to set `self._job_name` directly (#955), and my thinking that the job could see the value of `self._job_name` by reading the jobconf variable from the environment (#915).

Bottom line, `self._job_name` is just a confusing name. Something like `self._job_key` would avoid that confusion and point out that it's supposed to be unique.

**Comment by davidmarin** [comment]
Apparently there's an option that lets you set job_name, which is probably a bad idea.
